rocks/RocksStorage.java

```diff
@@ -185,7 +186,7 @@ public ByteArray get(ByteArray key) {
         }
 
         @Override
-        public <G> FunctionalIterator<G> iterate(ByteArray key, BiFunction<ByteArray, ByteArray, G> constructor) {
+        public <G extends Bytes.ByteComparable<G>> FunctionalIterator.Sorted<G> iterate(ByteArray key, BiFunction<ByteArray, ByteArray, G> constructor) {
```

Comment:
now anything we iterate from storage must be `ByteComparable`. We can `mapSorted` it later to convert it to other types as needed, all of which must be comparable as well